tests/models/test_restore.py

```diff
@@ -403,7 +438,7 @@ def assert_good_acc():
         tpipes.run_prediction(dp_model, dataloader, dp=True)
 
     # new model
-    model = EvalModelTemplate(**hparams)
+    model = BoringModel()
```

Comment:
Hi @rohitgr7, I'm looking at the errors in , but they're not directly related to GPUs. In particular, error ` FAILED tests/models/test_restore.py::test_dp_resume - TypeError: validation_step() missing 1 required positional argument: ` is related to missing `batch_idx` of the third validation batch when `new_trainer.fit(model)` is called:

```
args = (tensor([[ 0.3957,  0.3204,  1.2297,  1.7594,  2.4312,  1.0279, -0.5972, -0.2981,
          0.7191,  1.0529,  0.4432,  0.3232, -0.2949,  1.3755, -1.6502, -0.3943,
          0.5118, -0.1864,  1.4559,  0.7368,  0.1313, -0.2334, -0.7938, -0.5922,
          0.3301, -0.1205,  0.1629, -0.5484, -0.5139, -0.0191,  0.7912,  0.5872]],
       device='cuda:0'),)
```
First two batches have the batch_idx field. This is the second (idx 1):
```
args = (tensor([[ 0.8496, -0.5701,  1.2025,  1.9781,  0.3139,  0.3777,  0.6508,  0.4263,
          0.6376,  0.2734,  1.3930,  0.7251, -0.4903,  1.3228,  0.1101,  0.7859,
         -0.7761,  0.2667, -0.2144, -0.4566, -0.0672, -0.7441,  0.3575, -0.8824,
         -0.9212, -1.8361, -0.5334, -0.1589, -0.0142, -0.2120, -0.4227, -0.7254]],
       device='cuda:0'), 1)
```
I'm investigating, but it seems strange to me

Reply:
There's another issue, related to parallel gpus. 
When you call 
```
output = trained_model(batch)
```
if `trained_model` is of class `LightningDataParallel` it returns an output like:
```
{'x': tensor([1211.6818], device='cuda:0')}
```
This is nonsense, since the model should return two values when called `forward`, instead of a `dict`.
Do you have any idea?

Reply:
not sure about this. Still not very much comfortable with Lightning distributed code since I don't do distributed training often.

Reply:
ok.. dp test resolved. 2 to go.

Reply:
Did you solve this in the end?

Reply:
@gianscarpe only 1 test is failing now. As of now it's working with EvalModel but not with BoringModel. `test_amp_with_apex`

Reply:
we fixed it on master btw :)

Reply:
Perfect! :)